### PR TITLE
Missing and incorrect language keys

### PR DESCRIPTION
**Pull request by brianteeman**
- [x] BANNERS_SEARCH_IN_TITLE at: JROOT\administrator\templates\hathor\html\com_banners\banners\default.php:36
- [x] BANNERS_SELECT_TYPE at: JROOT\administrator\templates\hathor\html\com_banners\tracks\default.php:60
- [x] COM_CONTACT_SESSION_INVALID at: JROOT\components\com_contact\controllers\contact.php:64
- [x] COM_FINDER_INDEX_PURGE_FAILED at: JROOT\administrator\components\com_finder\controllers\index.php:58
- [x] COM_LANGS_SEARCH_IN_TITLE at: JROOT\administrator\templates\hathor\html\com_languages\languages\default.php:39
- [x] COM_TAGSS_REBUILD_FAILURE at: JROOT\administrator\components\com_tags\controllers\tags.php:62
- [x] COM_TAGS_DEFAULT_PAGE_TITLE at: JROOT\components\com_tags\views\tags\view.html.php:166
- [x] COM_USERS_FILTER_FILTER_DATE at: JROOT\administrator\templates\hathor\html\com_users\users\default.php:77
- [x] COM_USERS_REMIND_REQUEST_ERROR at: JROOT\components\com_users\controllers\user.php:392
- [x] COM_USER_LOGOUT_IMAGE_ALT at: JROOT\components\com_users\views\login\tmpl\default_logout.php:30
- [x] ERRORUNKOWNEXTENSION at: JROOT\libraries\cms\installer\adapter\library.php:367
- [x] FILTERS at: JROOT\administrator\templates\hathor\html\com_templates\templates\default.php:33
- [x] JERROR_SAVE_NOT_PERMITTED at: JROOT\administrator\components\com_templates\controllers\template.php:221
- [x] JFRAMEWORK_FORM_FIELDS_PLUGINS_ERROR_FOLDER_EMPTY at: JROOT\libraries\joomla\form\fields\plugins.php:144
- [x] JGLOBAL_BATCH_MOVE_ROW_NOT_FOUND at: JROOT\administrator\components\com_categories\models\category.php:1107
- [x] JGLOBAL_BATCH_MOVE_ROW_NOT_FOUND at: JROOT\administrator\components\com_categories\models\category.php:923
- [x] JGLOBAL_BATCH_MOVE_ROW_NOT_FOUND at: JROOT\administrator\components\com_menus\models\item.php:243
- [x] JGLOBAL_BATCH_MOVE_ROW_NOT_FOUND at: JROOT\administrator\components\com_menus\models\item.php:422
- [x] JGLOBAL_UNCATEGORISED at: JROOT\templates\beez3\html\com_content\featured\default_item.php:72
- [x] JGLOBAL_UNCATEGORISED at: JROOT\templates\beez3\html\com_content\featured\default_item.php:84
- [x] JLIB_APPLICATION_ERROR_MENU_LOAD at: JROOT\libraries\cms\menu\menu.php:130
- [x] JLIB_CLIENT_ERROR_JFTP_BAD_RESPONSE_SIZE at: JROOT\libraries\joomla\client\ftp.php:1351
- [x] JLIB_CLIENT_ERROR_JFTP_SYS_BAD_RESPONSE_NATIVE at: JROOT\libraries\joomla\client\ftp.php:447
- [x] JLIB_DATABASE_ERROR_DATABASE_QUERY at: JROOT\installation\model\database.php:339
- [x] JLIB_DATABASE_ERROR_DATABASE_QUERY at: JROOT\installation\model\database.php:356
- [x] JLIB_FILESYSTEM_PATCHER_ADD_LINE at: JROOT\libraries\joomla\filesystem\patcher.php:423
- [x] JLIB_FILESYSTEM_PATCHER_REMOVE_LINE at: JROOT\libraries\joomla\filesystem\patcher.php:413
- [x] JLIB_INSTALLER_ERROR_LIB_REFRESH_MANIFEST_CACHE at: JROOT\libraries\cms\installer\adapter\library.php:501
- [x] JLIB_INSTALLER_ERROR_PACK_REFRESH_MANIFEST_CACHE at: JROOT\libraries\cms\installer\adapter\file.php:634
- [x] JLIB_INSTALLER_ERROR_PACK_REFRESH_MANIFEST_CACHE at: JROOT\libraries\cms\installer\adapter\package.php:686
- [x] JLIB_INSTALLER_ERROR_TPL_REFRESH_MANIFEST_CACHE at: JROOT\libraries\cms\installer\adapter\template.php:606
- [x] JLIB_INSTALLER_INCORRECT_SEQUENCE at: JROOT\libraries\cms\installer\script.php:142
- [ ] JLIB_LOGIN_AUTHORISATION at: JROOT\libraries\cms\application\cms.php:850
- [ ] JLIB_LOGIN_AUTHORISATION at: JROOT\libraries\legacy\application\application.php:660
- [ ] JLIB_LOGIN_DENIED at: JROOT\libraries\cms\application\cms.php:845
- [ ] JLIB_LOGIN_DENIED at: JROOT\libraries\legacy\application\application.php:656
- [ ] JLIB_LOGIN_EXPIRED at: JROOT\libraries\cms\application\cms.php:840
- [ ] JLIB_LOGIN_EXPIRED at: JROOT\libraries\legacy\application\application.php:652
- [ ] JLIB_MAIL_MSG at: JROOT\libraries\joomla\mail\mail.php:757
- [ ] JLIB_MAIL_MSG_ADMIN at: JROOT\libraries\joomla\mail\mail.php:756
- [ ] JLIB_MAIL_USER_SUBMITTED at: JROOT\libraries\joomla\mail\mail.php:754
- [x] JLIB_STEMMER_INVALID_STEMMER at: JROOT\libraries\joomla\language\stemmer.php:58
- [x] JRULE_ALLOWED_INHERITED at: JROOT\libraries\cms\html\rules.php:227
- [x] JRULE_DENIED_INHERITED at: JROOT\libraries\cms\html\rules.php:228
- [x] LIB_CMS_WARNING_PROVIDE_VALID_NAME at: JROOT\libraries\cms\table\corecontent.php:97
- [x] RENAME FAILED at: JROOT\libraries\joomla\filesystem\folder.php:447
- [x] RENAME FAILED at: JROOT\libraries\joomla\filesystem\folder.php:456
- [ ] SOME_ERROR_CODE at: JROOT\administrator\components\com_config\model\application.php:177
- [ ] SOME_ERROR_CODE at: JROOT\administrator\components\com_config\model\application.php:209
- [ ] SOME_ERROR_CODE at: JROOT\administrator\components\com_config\model\application.php:402
- [x] UNKNOWN at: JROOT\libraries\cms\installer\installer.php:2183
- [x] UNKNOWN at: JROOT\libraries\cms\installer\installer.php:2184

removed the fof strings

### CODE REVIEW

**Comment by brianteeman** [comment]
JRULE_ALLOWED_INHERITED
JRULE_DENIED_INHERITED

I think this should be 
JLIB_RULES_ALLOWED_INHERITED="Allowed (Inherited)"
JLIB_RULES_DENIED_INHERITED="Denied (Inherited)"

**Comment by brianteeman** [comment]
JLIB_LOGIN_EXPIRED 
I think this should be changed to use this key
JLIB_ENVIRONMENT_SESSION_EXPIRED="Your session has expired. Please log in again."

**Comment by brianteeman** [comment]
JLIB_LOGIN_DENIED
I think this should be changed to use this key
JERROR_LOGIN_DENIED="You do not have access to the Administrator section of this site."

**Comment by infograf768** [comment]
from ipad
RENAME is wrong. we are no more in the streams conditional afaik

File: administrator/language/en-GB/en-GB.lib_joomla.ini [hunk 1]
```diff
@@ -67,81 +68,82 @@ JLIB_CACHE_ERROR_CACHE_STORAGE_LOAD="Unable to load Cache Storage: %s"
...
+JLIB_CLIENT_ERROR_JFTP_STORE_READING_LOCAL="JFTP: :store: Unable to open local file for reading. Local path: %s"
+JLIB_CLIENT_ERROR_JFTP_SYST_BAD_RESPONSE="JFTP: :syst: Bad response. Server response: %s [Expected: 215]"
+JLIB_CLIENT_ERROR_JFTP_SYST_BAD_RESPONSE_NATIVE="JFTP: :syst: Bad response."
+JLIB_CLIENT_ERROR_JFTP_VERIFYRESPONSE="JFTP: :_verifyResponse: Timeout or unrecognised response while waiting for a response from the server. Server response: %s"
+JLIB_CLIENT_ERROR_JFTP_WRITE_BAD_RESPONSE="JFTP: :write: Bad response."
+JLIB_CLIENT_ERROR_JFTP_WRITE_BAD_RESPONSE_STOR="JFTP: :write: Bad response. Server response: %1$s [Expected: 150 or 125]. Path sent: %2$s"
+JLIB_CLIENT_ERROR_JFTP_WRITE_BAD_RESPONSE_TRANSFER="JFTP: :write: Transfer Failed. Server response: %1$s [Expected: 226]. Path sent: %2$s"
+JLIB_CLIENT_ERROR_JFTP_WRITE_DATA_PORT="JFTP: :write: Unable to write to data port socket."
+JLIB_CLIENT_ERROR_JFTP_WRITE_PASSIVE="JFTP: :write: Unable to use passive mode."
+s
```
Comment (zero-24) [inline administrator/language/en-GB/en-GB.lib_joomla.ini]: Is this extra s expected?

Comment (brianteeman) [inline administrator/language/en-GB/en-GB.lib_joomla.ini]: Looks like an error I will check later

**Comment by zero-24** [comment]
- STATUS at: JROOT\libraries\fof\utils\installscript\installscript.php:553
- EXTENSION at: JROOT\libraries\fof\utils\installscript\installscript.php:552

Both are issues with the unsupported upstream repo.

**Comment by brianteeman** [comment]
@zero-24 yes but that doesnt mean we cant fix them here. Just not sure how

**Comment by zero-24** [comment]
We also exclude external libs from CS rules 😃  but it is above my pay grade to decide it.

**Comment by brianteeman** [comment]
And mine

File: libraries/joomla/filesystem/folder.php [hunk 2]
```diff
@@ -444,7 +444,7 @@ public static function move($src, $dest, $path = '', $use_streams = false)
 				// Use FTP rename to simulate move
 				if (!$ftp->rename($src, $dest))
 				{
-					return JText::_('Rename failed');
+					return JText::sprintf('JLIB_FILESYSTEM_ERROR_FOLDER_RENAME', $stream->getError());
```
Comment (infograf768) [inline libraries/joomla/filesystem/folder.php]: I am not sure of this.

File: libraries/joomla/filesystem/folder.php [hunk 3]
```diff
@@ -453,7 +453,7 @@ public static function move($src, $dest, $path = '', $use_streams = false)
 			{
 				if (!@rename($src, $dest))
 				{
-					return JText::_('Rename failed');
+					return JText::sprintf('JLIB_FILESYSTEM_ERROR_FOLDER_RENAME', $stream->getError());
```
Comment (infograf768) [inline libraries/joomla/filesystem/folder.php]: Same

**Comment by infograf768** [comment]
JLIB_DATABASE_ERROR_DATABASE_QUERY only concerns installation (which is frozen)
Should be something like
INSTL_DATABASE_ERROR_POSTGRESQL_QUERY="PostgreSQL database query failed."
to add after line 193

**Comment by infograf768** [comment]
Proposal for
JLIB_INSTALLER_INCORRECT_SEQUENCE

JLIB_INSTALLER_INCORRECT_SEQUENCE="Downgrading from version %1$s to version %2$s is not allowed."

**Comment by brianteeman** [comment]
Can I change that to "not possible"

File: libraries/joomla/filesystem/folder.php (hunk 2 above)
Comment (sovainfo) [inline libraries/joomla/filesystem/folder.php]: Suggest:
`return JText::sprintf('JLIB_FILESYSTEM_ERROR_FOLDER_RENAME', 'using FTP');`

File: libraries/joomla/filesystem/folder.php (hunk 3 above)
Comment (sovainfo) [inline libraries/joomla/filesystem/folder.php]: Suggest:
`return JText::sprintf('JLIB_FILESYSTEM_ERROR_FOLDER_RENAME', 'using PHP');`

Comment (brianteeman) [inline libraries/joomla/filesystem/folder.php]: @sovainfo but that isn't translatable?

Comment (sovainfo) [inline libraries/joomla/filesystem/folder.php]: Just thought of that, you could leave out ` using`

**Comment by infograf768** [comment]
>Can I change that to "not possible"

Not sure as it seems that it depends on `$allowDowngrades` which could be used by a 3rd party preflight.
Because of
`if (strtolower($type) == 'update' && !$this->allowDowngrades)`

**Comment by brianteeman** [comment]
I will leave the two strings that are part of the installer to a separate PR as installation is frozen

File: libraries/joomla/filesystem/folder.php (hunk 3 above)
Comment (brianteeman) [inline libraries/joomla/filesystem/folder.php]: @infograf768 thoughts? suggestions?

**Comment by brianteeman** [comment]
The installation string changes can be found in #10816 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10804">issues.joomla.org/joomla-cms/10804</a>.</sub>

**Comment by infograf768** [comment]
@brianteeman 
Can you solve the conflicts here? Otherwise I am afraid this will have to be postponed to 3.6.1

**Comment by brianteeman** [comment]
I will look at the conflicts this morning<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10804">issues.joomla.org/joomla-cms/10804</a>.</sub>

**Comment by brianteeman** [comment]
Done as much as I can

**Comment by infograf768** [comment]
let's wait for the acl pr being merged as it looks like the jrule strings are useless. maybe as well for the code using them. in any case they should not be named that way.

**Comment by brianteeman** [comment]
resolved conflicts

File: administrator/templates/hathor/html/com_banners/tracks/default.php [hunk 4]
```diff
@@ -28,6 +28,107 @@
 <?php else : ?>
 	<div id="j-main-container">
 <?php endif;?>
+<<<<<<< HEAD
```
Comment (andrepereiradasilva) [inline administrator/templates/hathor/html/com_banners/tracks/default.php]: ?

File: administrator/templates/hathor/html/com_banners/tracks/default.php [hunk 5]
```diff
@@ -28,6 +28,107 @@
...
+
+	<?php echo $this->pagination->getListFooter(); ?>
+
+	<input type="hidden" name="task" value="" />
+	<input type="hidden" name="boxchecked" value="0" />
+	<input type="hidden" name="filter_order" value="<?php echo $listOrder; ?>" />
+	<input type="hidden" name="filter_order_Dir" value="<?php echo $listDirn; ?>" />
+	<?php echo JHtml::_('form.token'); ?>
+</div>
+=======
```
Comment (andrepereiradasilva) [inline administrator/templates/hathor/html/com_banners/tracks/default.php]: ?

File: administrator/templates/hathor/html/com_banners/tracks/default.php [hunk 6]
```diff
@@ -146,4 +247,5 @@
 		<input name="filter_order_Dir" type="hidden" value="<?php echo $listDirn; ?>" />
 		<?php echo JHtml::_('form.token'); ?>
 	</div>
+>>>>>>> edb70c27612c0bdedca70044877cfef030e2a656
```
Comment (andrepereiradasilva) [inline administrator/templates/hathor/html/com_banners/tracks/default.php]: ?

File: administrator/templates/hathor/html/com_banners/tracks/default.php (hunk 4 above)
Comment (brianteeman) [inline administrator/templates/hathor/html/com_banners/tracks/default.php]: grrrh

**Comment by brianteeman** [comment]
I am going to close this and redo it

**Comment by brianteeman** [comment]
Closed see #11611

**Pull request closed without merging**